src/cf/client.go

```diff
@@ -186,6 +199,21 @@ func (c *cfClient) GetTokens() Tokens {
 	return c.tokens
 }
 
+func (c *cfClient) GetTokensWithRefresh() Tokens {
+	if c.isTokenToBeExpired() {
```

Comment:
this is used to avoid another  refresh if other thread has already refreshed the token and changed the expire time.

We can remove the outer check while this will have more overhead since  no matter the token is to be expired or not we need lock/unlock